firebase-auth/src/commonMain/kotlin/dev/gitlive/firebase/auth/multifactor.kt

```diff
@@ -0,0 +1,34 @@
+/*
+ * Copyright (c) 2020 GitLive Ltd.  Use of this source code is governed by the Apache 2.0 license.
+ */
+
+package dev.gitlive.firebase.auth
+
+expect class MultiFactor {
+    val enrolledFactors: List<MultiFactorInfo>
+    suspend fun enroll(multiFactorAssertion: MultiFactorAssertion, displayName: String?)
+    suspend fun getSession(): MultiFactorSession
+    suspend fun unenroll(multiFactorInfo: MultiFactorInfo)
+    suspend fun unenroll(factorUid: String)
+}
+
+expect class MultiFactorInfo {
+    val displayName: String?
+    val enrollmentTime: Long
+    val factorId: String
+    val uid: String
+}
+
+expect class MultiFactorAssertion {
+    val factorId: String
+}
```

Comment:
why do you think they wrap factorId in a class instead of just using factorId in enroll?

Reply:
Well on Android MultiFactorAssertion is an abstract class that has a subclass for PhoneMultiFactorAssertions. So I'm guessing this is future proofed for additional Multifactor solutions. However it seems to be a relatively new addition to Firebase and does not seem to fully cover all platforms yet. So I'd consider multifactor somewhat WIP for now.